Added analog stick to gamepad shield

sh3raawii opened the pull request:
Added a mode to the gamepad shield so that the users can switch between Keys and Analog

Commit:
Merge remote-tracking branch 'Integreight/development' into Joystick_Shield

Commit:
Merge remote-tracking branch 'Integreight/development' into Joystick_Shield

Commit:
Added an Analog Stick to the gamepad

added a new mode to the gamepad to allow users to switch between keys
and analog

Commit:
Merge remote-tracking branch 'Integreight/development' into Joystick_Shield

alieslam commented:
Great work .. just two points to consider: 
1- Function calcDirection has a typo stated as "clacDirection"
2- Highlight the following "CENTER-RIGHT ... " enum as LITERALS and getPower, getDirection, getAngle, getMode as functions in Keyword.txt

That's it for now nice job.. ;)

Commit:
Fix a typo in GamePad Shield

Commit:
Merge remote-tracking branch 'Integreight/development' into Joystick_Shield

Commit:
Highlight Gamepad's added functions and Keywords

Here is a List of added Functions:
getMode, getPower, getAngle, getDirection
Direction Enum Keywords:
CENTER, RIGHT, UPPER_RIGHT, UP, UPPER_LEFT, LEFT, LOWER_LEFT, DOWN,
LOWER_RIGHT

sh3raawii committed:
Empty Commit
